From f09f47b63a0a079c20e8152d5c11e5f44848475f Mon Sep 17 00:00:00 2001
From: Liwei Song <liwei.song@windriver.com>
Date: Sun, 16 Aug 2015 11:11:31 +0800
Subject: [PATCH 2/5] fix the compile error led by __vlan_hwaccel_put_tag
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

fix the follwing compile error:
r8168_n.c:2062:3: error: too few arguments to function ‘__vlan_hwaccel_put_tag’

since enable CONFIG_R8168_NAPI and CONFIG_R8168_VLAN it will import this error.
without this two config enable, r8168 will not use __vlan_hwaccel_put_tag().
this is due to the __vlan_hwaccel_put_tag in wr has three args, but the old r8168
just has two, so add the third args just like the new r8168 driver did.

Signed-off-by: Liwei Song <liwei.song@windriver.com>
---
 drivers/net/ethernet/realtek/r8168_n.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/net/ethernet/realtek/r8168_n.c b/drivers/net/ethernet/realtek/r8168_n.c
index 8862f67..b8026e7 100644
--- a/drivers/net/ethernet/realtek/r8168_n.c
+++ b/drivers/net/ethernet/realtek/r8168_n.c
@@ -2059,7 +2059,7 @@ rtl8168_rx_vlan_skb(struct rtl8168_private *tp,
 	}
 #else
 	if (opts2 & RxVlanTag)
-		__vlan_hwaccel_put_tag(skb, swab16(opts2 & 0xffff));
+		__vlan_hwaccel_put_tag(skb, htons(ETH_P_8021Q), swab16(opts2 & 0xffff));
 #endif
 
 	desc->opts2 = 0;
-- 
1.7.5.4

